plugins/modules/ndo_mcp_global_policy.py

```diff
@@ -0,0 +1,341 @@
+        admin_state=dict(type="str", choices=["enabled", "disabled"]),
+        key=dict(type="str", no_log=False),
+        per_vlan=dict(
+            type="str",
+            aliases=["per_epg", "mcp_pdu_per_vlan"],
+            choices=["enabled", "disabled"],
+        ),
+        loop_detection_factor=dict(type="int", aliases=["loop_factor", "loop_detection_mult_factor"]),
+        port_disable=dict(
+            type="str",
+            aliases=["port_disable_protection"],
+            choices=["enabled", "disabled"],
+        ),
+        initial_delay_time=dict(type="int", aliases=["initial_delay"]),
+        transmission_frequency_sec=dict(type="int", aliases=["tx_freq"]),
+        transmission_frequency_msec=dict(type="int", aliases=["tx_freq_ms"]),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    # Enforcing that a user must specify a name or uuid when
+    # adding, updating or removing even though there is only one policy per template.
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    state = module.params.get("state")
+    per_vlan = module.params.get("per_vlan")
+    port_disable = module.params.get("port_disable")
+
+    mso_values = {
+        "name": name,
+        "description": module.params.get("description"),
+        "adminState": module.params.get("admin_state"),
+        "key": module.params.get("key"),
+        "enablePduPerVlan": ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP[per_vlan] if per_vlan else None,
+        "loopDetectMultFactor": module.params.get("loop_detection_factor"),
+        "protectPortDisable": ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP[port_disable] if port_disable else None,
+        "initialDelayTime": module.params.get("initial_delay_time"),
+        "txFreq": module.params.get("transmission_frequency_sec"),
+        "txFreqMsec": module.params.get("transmission_frequency_msec"),
+    }
+
+    ops = []
+
+    mso_template = MSOTemplate(mso, "fabric_policy", module.params.get("template"))
+    mso_template.validate_template("fabricPolicy")
+
+    object_description = "MCP Global Policy"
+    object_base_path = "/fabricPolicyTemplate/template/mcpGlobalPolicy"
+
+    existing_mcp_global_policy = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("mcpGlobalPolicy", {})
+
+    if state in ["query", "absent"] and existing_mcp_global_policy == {}:
+        mso.exit_json()
+
+    elif state == "query" and not (name or uuid):
+        mso.existing = [existing_mcp_global_policy]
+
+    elif existing_mcp_global_policy and (name or uuid):
+        mso.existing = mso.previous = copy.deepcopy(existing_mcp_global_policy)
+
```

Comment:
Would it be possible to add check_existing_identifier(mso, object_description, uuid) here, so that we don't have to call it twice? We could even forego the separate function for this if that's possible

Reply:
Yes, this is possible. Moved the check logic and removed the function as its only needed in one place now.